app/src/pages/outside/loginPage/pageBlocks/loginBlock/loginBlock.jsx

```diff
@@ -29,8 +34,30 @@ const messages = defineMessages({
   },
 });
 
-export const LoginBlock = () => (
-  <PageBlockContainer header={messages.welcome} hint={messages.login}>
-    <LoginForm />
-  </PageBlockContainer>
-);
+@connect(
+  (state) => ({
+    extensions: uiExtensionLoginPageSelector(state),
+  }),
+  {},
+)
+export class LoginBlock extends PureComponent {
+  static propTypes = {
+    extensions: PropTypes.arrayOf(extensionType),
+  };
+  static defaultProps = {
+    extensions: [],
+  };
+
+  render() {
+    const { extensions } = this.props;
+
+    return (
+      <PageBlockContainer header={messages.welcome} hint={messages.login}>
+        <LoginForm />
+        {extensions.map((extension) => (
+          <ExtensionLoader extension={extension} />
+        ))}
```

Comment:
According to the layout you need to render extensions before (under) `LoginForm` (see the designs).

Reply:
Also please note that connected components already pure (see [react-redux](https://react-redux.js.org/api/connect) docs).
As we use React 16+ we prefer to write new components as functional using React hooks, so you can use [use-selector](https://react-redux.js.org/api/hooks#useselector) to extract the extensions inside of functional component.